### DISCUSSION TITLE
Monitoring for PostgreSQL HA cluster

### DISCUSSION THREAD

**Discussion by Alexanator** [autobase-tech/autobase#212] [Q&A]
Hi, Community!

Is there any solutions to monitor PostgreSQL cluster and state of all its tools? Like patroni, haproxy, etcd, keepalived, etc.
I am using[ potgres-exporter ](https://github.com/prometheus-community/postgres_exporter) to monitor my databases but it provides metrics for postgresql only.

**Comment by vitabaks** [comment]
@Alexanator I use **Zabbix** to monitor cluster components.

Example of Zabbix templates:

- [PostgreSQL_Patroni_Cluster_TypeA.xml.zip](https://github.com/vitabaks/postgresql_cluster/files/9913114/PostgreSQL_Patroni_Cluster_TypeA.xml.zip)
- [PostgreSQL_Patroni_Cluster_TypeB.xml.zip](https://github.com/vitabaks/postgresql_cluster/files/9913115/PostgreSQL_Patroni_Cluster_TypeB.xml.zip)

**Reply by vitabaks** [reply]
For PostgreSQL monitoring I use **pgwatch2 - Postgres.ai Edition** - https://gitlab.com/postgres-ai/pgwatch2

To monitor the system, I use **Netdata** - https://github.com/netdata/netdata
